### DISCUSSION TITLE
problema con il database

### DISCUSSION THREAD

**Discussion by fabriziotoppi** [iisgiua/giuaschool#377] [Richieste di aiuto]
Salve, non riesco a ultimare l'installazione . il processo si blocca subito dopo aver inserito i dati del databaseweb . 

allego errore

![Schermata 2022-09-27 alle 16 47 46](https://user-images.githubusercontent.com/113930669/192559250-870e1b2b-9002-4cab-93e1-96e7dea430af.png)
![Schermata 2022-09-27 alle 16 47 37](https://user-images.githubusercontent.com/113930669/192559260-29e9aa62-fe10-4cdd-af15-a3b4099e8e42.png)

**Comment by trinko** [comment]
Per il primo errore, controlla che il file **".env"**, presente nella cartella principale, contenga una riga del tipo:
```
APP_SECRET='CodiceDaCambiareConUnoPiùComplesso'
```

Il secondo errore sembra sia dovuto a dei limiti del tuo servizio di hosting, che non permette alcune operazioni sul database.
Si può aggirare il problema inserendo a mano i comandi SQL per la creazione delle tabelle, ma devi avere almeno accesso al database con uno strumento come **phpMyAdmin**. Se è così ti fornisco il file dei comandi SQL.

**Reply by fabriziotoppi** [reply]
il servizio di hosting è quello di aruba con database mysql. Penso sia standard

**Reply by fabriziotoppi** [reply]
app-segret. presente con un codice alfanumerico

**Comment by trinko** [comment]
Molto strano, ma sei sicuro di aver utilizzato la [versione 1.5.0](https://github.com/iisgiua/giuaschool/releases/download/v1.5.0/giuaschool-release-v1.5.0.zip)? 

Non è che hai scaricato gli aggiornamenti successivi, che non sono ancora testati per l'installazione?

**Comment by fabriziotoppi** [comment]
ho fatto una nuova cartella e lanciato il file index.php . modificato il
file .end con password  ma non riesco ad aggancairlo.
ma il codice app secret va cambiato?

Il giorno mer 28 set 2022 alle ore 14:23 Antonello Dessì <
***@***.***> ha scritto:

**Comment by trinko** [comment]
Hai imposto l'uso del PHP 7.4 o successivo? 
Questo può essere causa di parecchi problemi.

**Comment by fabriziotoppi** [comment]
è impostato su 8.0

Il giorno gio 29 set 2022 alle ore 17:36 Antonello Dessì <
***@***.***> ha scritto:

**Comment by fabriziotoppi** [comment]
guarda ho provato anche a utilizzare il localhost  ma nulla da fare.
Mi dispiace ma abbandono un progetto molto interessante e notevole.
Seguiro sicuramente l'evoluzione, magari piu in la ci riproveremo.
Ti ringrazio per la tua disponibilità
Fabrizio


Il giorno gio 29 set 2022 alle ore 18:00 fabrizio toppi <
***@***.***> ha scritto:

**Comment by zerai** [comment]
@fabriziotoppi

> magari piu in la ci riproveremo

é molto probabile che l'errore sul db sia dovuto alla non corretta configurazione (utente mysql <-> db mysql)
nel pannello mysql aruba dovresti trovare l'utente con il quale accedi al db e verificare che oltre ad avere il set minimo di permessi select|insert|update|delete ci siano anche i permessi create|alter che sono quelli usati per creare e modificare le tabelle.
